contracts/cross-chain/SiphonReceiver.sol

```diff
@@ -0,0 +1,166 @@
+ *      When rewardClaimed is called on the Booster
+ */
+contract SiphonReceiver is ILayerZeroReceiver, Ownable {
+    using SafeERC20 for IrCvx;
+
+    /* -------------------------------------------------------------------
+      Storage 
+    ------------------------------------------------------------------- */
+
+    /// @dev Booster contract
+    address public booster;
+
+    /// @dev rCVX contract
+    IrCvx public rCvx;
+
+    /// @dev Siphon depositor on L1
+    address public l1SiphonDepositor;
+
+    /// @dev Layer Zero endpoint
+    ILayerZeroEndpoint public lzEndpoint;
+
+    /// @dev Destination chain ID used by Layer Zero
+    uint16 public immutable dstChainId;
+
+    /* -------------------------------------------------------------------
+      Events 
+    ------------------------------------------------------------------- */
+
+    event UpdateBooster(address sender, address booster);
+
+    event Mint(address sender, address to, uint256 amount);
+
+    event Convert(address sender, uint256 amount, bool lock);
+
+    /* -------------------------------------------------------------------
+      Constructor 
+    ------------------------------------------------------------------- */
+
+    constructor(
+        IrCvx _rCvx,
+        address _l1SiphonDepositor,
+        ILayerZeroEndpoint _lzEndpoint,
+        uint16 _dstChainId
+    ) {
+        lzEndpoint = _lzEndpoint;
+        l1SiphonDepositor = _l1SiphonDepositor;
+        rCvx = _rCvx;
+        dstChainId = _dstChainId;
+    }
+
+    /* -------------------------------------------------------------------
+      Setter functions 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @dev Set the Booster address
+     * @param _booster Booster address
+     */
+    function setBooster(address _booster) external onlyOwner {
+        booster = _booster;
+        emit UpdateBooster(msg.sender, _booster);
+    }
+
+    /* -------------------------------------------------------------------
+      Core functions 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @dev Mint function called by Booster.rewardClaimed. rCVX tokens are
+     *      minted on the L1 chain and sent to this contract on the L2 chain
+     *      when rewardClaimed is called on the booster rCVX tokens are sent
+     *      to the sender.
+     * @param _to     Address to send rCvx to
+     * @param _amount Amount of rCvx to send
+     */
+    function mint(address _to, uint256 _amount) external {
+        require(msg.sender == booster, "!booster");
+        rCvx.safeTransfer(_to, _amount);
+        emit Mint(msg.sender, _to, _amount);
+    }
+
+    /**
+     * @dev The Booster calls this function when earmarking rewards.
+     *      Instead of having to make changes to the Booster we just
+     *      implement the method interface so it can be called silently
+     */
+    function queueNewRewards(uint256) external {
```

Comment:
I think we should just remove those lines from the booster. There is lots that can be removed from it tbh